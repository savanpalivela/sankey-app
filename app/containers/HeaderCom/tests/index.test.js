/**
 *
 * Tests for HeaderCom
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import ShallowRenderer from 'react-test-renderer/shallow';

import { HeaderCom } from '../headerCom.container';

const renderer = new ShallowRenderer();

describe('<HeaderCom />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<HeaderCom changeLocaleHandler={() => {}} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    renderer.render(<HeaderCom changeLocaleHandler={() => {}} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
