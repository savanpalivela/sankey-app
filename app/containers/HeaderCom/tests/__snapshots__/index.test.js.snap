// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderCom /> should render and match the snapshot 1`] = `
<div
  className="undefined row"
>
  <div
    className="col-sm-2"
  >
    <img
      alt="logo"
      src="IMAGE_MOCK"
    />
  </div>
  <div
    className="col-sm-8"
  />
  <div
    className="undefined col-sm-2"
  >
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
      EN
    </button>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
      DE
    </button>
  </div>
</div>
`;
